public/Disable-DbaReplDistributor.ps1

```diff
@@ -0,0 +1,102 @@
+
+    .PARAMETER Confirm
+        If this switch is enabled, you will be prompted for confirmation before executing any operations that change state.
+
+    .NOTES
+        Tags: repl, Replication
+        Author: Jess Pomfret (@jpomfret), jesspomfret.com
+
+        Website: https://dbatools.io
+        Copyright: (c) 2023 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Disable-DbaReplDistributor
+
+    .EXAMPLE
+        PS C:\> Disable-DbaReplDistributor -SqlInstance mssql1
+
+        Disables replication distribution for the mssql1 instance.
+
+    .EXAMPLE
+        PS C:\> $cred = Get-Credential sqladmin
+        PS C:\> Disable-DbaReplDistributor -SqlInstance mssql1, mssql2 -SqlCredential $cred -Force
+
+        Disables replication distribution for the mssql1 and mssql2 instances using a sql login.
+        Specifies force so the publishing and Distributor configuration at the current server is uninstalled
+        regardless of whether or not dependent publishing and distribution objects are uninstalled.
+
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [Parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [switch]$Force,
+        [switch]$EnableException
+    )
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $replServer = Get-DbaReplServer -SqlInstance $instance -SqlCredential $SqlCredential -EnableException:$EnableException
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+            Write-Message -Level Verbose -Message "Disabling replication distribution for $instance"
+
+            if ($replServer.IsDistributor) {
+                try {
+                    if ($PSCmdlet.ShouldProcess($instance, "Disabling distribution on $instance")) {
+                        # remove any connections to the distribution database
+                        Get-DbaProcess -SqlInstance $instance -SqlCredential $SqlCredential -Database $replServer.DistributionDatabases.name -EnableException:$EnableException | Stop-DbaProcess -EnableException:$EnableException
+                        # uninstall distribution
+                        $replServer.UninstallDistributor($Force)
+                    }
+                } catch {
+                    Stop-Function -Message "Unable to disable replication distribution" -ErrorRecord $_ -Target $instance -Continue
+                }
+
+                $replServer.Refresh()
+                $replServer
+
+            } else {
+                Stop-Function -Message "$instance isn't currently enabled for distributing." -ErrorRecord $_ -Target $instance -Continue
```

Comment:
What error would get captured here in `$_`? How is this message reached if I use `-Force`?